Automattic/wp-calypso
# Hosting Command Palette: Restructure the logic execution for the nested commands

Related to this comment on https://github.com/Automattic/wp-calypso/pull/85228#pullrequestreview-1784768660 

## Proposed Changes

* This PR refactors the code to make it more efficient for when the nested commands (site picker) and top-level commands are executed.

## Testing Instructions

* Navigate to Calypso Live Link
* Press `cmd + k` on macOS or `ctrl + k` on Windows to open the command palette
* Select `Open sites dashboard` command
* Select a site to navigate to
* Confirm that the command palette executes as expected
* Press `cmd + k` on macOS or `ctrl + k` on Windows to open the command palette from site's dashboard where you previously navigate
* Observe that the first command is `View my sites` (this is to confirm that the sorting of the top-level commands still works as expected)
* Generally compare the hosting palette in production and in this PR and confirm that they work in the same way e.g. the commands come in the same order both in production and in this PR

## Pre-merge Checklist
